ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/validator/StatementValidator.java

```diff
@@ -28,13 +29,29 @@
  */
 public class StatementValidator implements Validator {
 
+    private Validator nextInValidator;
+
+    public StatementValidator() {
+        nextInValidator = null;
+    }
+
     @Override
     public void setNextValidator(Validator nextValidator) {
+        this.nextInValidator = nextValidator;
     }
 
     @Override
     public boolean evaluate(String source) {
         IncompleteInputFinder incompleteInputFinder = new IncompleteInputFinder();
-        return NodeParser.parseStatement(source).apply(incompleteInputFinder);
+        Node parsedNode = NodeParser.parseBlockStatement("{" + source + "}");
+        if (!parsedNode.hasDiagnostics()) {
+            return true;
+        }
+
+        if (!parsedNode.apply(incompleteInputFinder)) {
+            return true;
+        }
+
+        return nextInValidator.evaluate(source);
```

Comment:
We can merge these together

Reply:
Updated in 7d25e1dea188151fc8c5f84a1bf9156ce22ebc81